rpc/api_addtransaction_test.go

```diff
@@ -0,0 +1,73 @@
+package rpc
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"testing"
+
+	"github.com/dontpanicdao/caigo/types"
+)
+
+//go:embed tests/counter.json
+var counterFile []byte
+
+// TestAddDeployTransaction tests AddDeployTransaction
+func TestAddDeployTransaction(t *testing.T) {
+	testConfig := beforeEach(t)
+	defer testConfig.client.Close()
+
+	type testSetType struct {
+		Salt                    string
+		Contract                types.ContractClass
+		ConstructorCallData     []string
+		ExpectedTransactionHash string
+		ExpectedContractAddress string
+	}
+	var contract types.ContractClass
+	err := json.Unmarshal(counterFile, &contract)
+	if err != nil {
+		t.Fatal("error loading contract:", err)
+	}
```

Comment:
nit: limit err scope
```suggestion
	if err := json.Unmarshal(counterFile, &contract); err != nil {
		t.Fatal("error loading contract:", err)
	}
```